dali/base/dautils.cpp

```diff
@@ -924,6 +924,36 @@ StringBuffer &CDfsLogicalFileName::makeXPathLName(StringBuffer &lfnNodeName) con
     return lfnNodeName;
 }
 
+static bool begins(const char *&ln,const char *pat)
+{
+    size32_t sz = strlen(pat);
+    if (memicmp(ln,pat,sz)==0) {
+        ln += sz;
+        return true;
+    }
+    return false;
+}
+
+StringBuffer &CDfsLogicalFileName::xPathToLName(const char *xpath, StringBuffer &lName) const
+{
+    const char *s = xpath;
+    if (!begins(s, "/Files"))
+        return lName;
+
+    while (*s && (begins(s, "/Scope[@name=\"") || begins(s, "/File[@name=\"") || begins(s, "/SuperFile[@name=\"")))
+    {
+        if (lName.length())
```

Comment:
would be better to use a flag to check if started.. as someone using this function may already have content in lName.